mysql/resource_database.go

```diff
@@ -116,20 +127,24 @@ func ReadDatabase(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	d.Set("name", name)
 	d.Set("default_character_set", defaultCharset)
 	d.Set("default_collation", defaultCollation)
 
 	return nil
 }
 
 func DeleteDatabase(d *schema.ResourceData, meta interface{}) error {
-	db := meta.(*providerConfiguration).DB
+	db, err := connectToMySQL(meta.(*MySQLConfiguration).Config)
+	if err != nil {
+		return err
+	}
 
 	name := d.Id()
 	stmtSQL := "DROP DATABASE " + quoteIdentifier(name)
 	log.Println("Executing statement:", stmtSQL)
 
-	_, err := db.Exec(stmtSQL)
+	_, err = db.Exec(stmtSQL)
 	if err == nil {
 		d.SetId("")
```

Comment:
FWIW this is no longer necessary in a delete

Reply:
Good catch. 👍